File: 003/bible.py
import codecs


def read_in_chunks(file_object, chunk_size=1024):
    while True:
        data = file_object.read(chunk_size)
        if not data:
            break
        yield data


def main():
    charcount = 0
    f = codecs.open('/tmp/bible.txt', encoding='utf-8')
    for chunk in read_in_chunks(f):
        charcount += len(chunk)
    print("There are {} characters in the bible".format(charcount))


if __name__ == '__main__':
    main()

File: 003/bible.c
#include <stdio.h>
#include <string.h>

#define BIBLE_PATH "/tmp/bible.txt"
#define BUFFER_SIZE 1024

int main(void)
{
  int charcount = 0;
  char buffer[BUFFER_SIZE];
  FILE *fp;

  if ((fp = fopen(BIBLE_PATH, "r")) == NULL)
  {
    printf("Cannot find path to bible");
    return 1;
  }

  while (fgets(buffer, BUFFER_SIZE, fp) != NULL)
    charcount += (int) strlen(buffer);

  printf("There are %d characters in the bible", charcount);
  return 0;
}
